### PR TITLE
fixes for numpy2

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: fastparquet/test/test_json.py [hunk 1]
```diff
@@ -25,15 +26,15 @@ def _clear_cache():
...
-        [1, 1, 2, 3, 5],
-        [1.23, -3.45],
-        [np.float64(0.12), np.float64(4.56)],
-        [[1, 2, 4], ["x", "y", "z"]],
-        {"k1": "value", "k2": "à/è", "k3": 3},
-        {"k1": [1, 2, 3], "k2": [4.1, 5.2, 6.3]},
+        (None, False),
+        ([1, 1, 2, 3, 5], False),
+        ([1.23, -3.45], False),
+        ([np.float64(0.12), np.float64(4.56)], True),
```
Comment (graingert) [inline fastparquet/test/test_json.py:34]: not sure what to do about this but rapidjson.dumps produces invalid json for:

```python
>>> rapidjson.dumps(np.float64(0.1))
'np.float64(0.1)'
```

**Comment by graingert** [comment]
Pandas build failing I think because of https://github.com/cython/cython/issues/5748

File: fastparquet/test/test_json.py (hunk 1 above)
Comment (martindurant) [reply fastparquet/test/test_json.py:34]: I don't suppose there's a good reason rapidjson should know what to do with that

**Comment by martindurant** [comment]
Thank you

**Pull request merged**